src/test/python/complete_winml_tutorial.py

```diff
@@ -0,0 +1,142 @@
+import find_projection
+
+import pyrt
+import asyncio
+
+def timed_op(fun):
+    import time
+
+    def sync_wrapper(*args, **kwds):
+        print("Starting", fun.__name__)
+        start = time.perf_counter()
+
+        ret = fun(*args, **kwds)
+
+        end = time.perf_counter()
+        print(fun.__name__, "took", end - start, "seconds")
+        return ret
+
+    async def async_wrapper(*args, **kwds):
+        print("Starting", fun.__name__)
+        start = time.perf_counter()
+
+        ret = await fun(*args, **kwds)
+
+        end = time.perf_counter()
+        print(fun.__name__, "took", end - start, "seconds")
+        return ret
+
+    return async_wrapper if asyncio.iscoroutinefunction(fun) else sync_wrapper
+
+async def wrap_async_op(op):
+    loop = asyncio.get_event_loop()
+    future = loop.create_future()
+
+    def callback(operation, status):
+        if status == 1:
+            result = operation.GetResults()
+            loop.call_soon_threadsafe(asyncio.Future.set_result, future, result)
+        elif status == 2:
+            loop.call_soon_threadsafe(asyncio.Future.set_exception, future, asyncio.CancelledError())
+        elif status == 3:
+            loop.call_soon_threadsafe(asyncio.Future.set_exception, future, RuntimeError("AsyncOp failed"))
+        else:
+            loop.call_soon_threadsafe(asyncio.Future.set_exception, future, RuntimeError("Unexpected AsyncStatus"))
+
+    op.put_Completed(callback)
+
+    return await future
+
+def run_async_code(code):
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(code())
+    loop.close()
+
+import pyrt.windows.ai.machinelearning as winml 
+import os
+
+@timed_op
+def load_model(model_path):
+    return winml.LearningModel.LoadFromFilePath(os.fspath(model_path))
+
+@timed_op
+async def load_image_file(file_path):
+    from pyrt.windows.storage import StorageFile
+    from pyrt.windows.graphics.imaging import BitmapDecoder
+    from pyrt.windows.media import VideoFrame
+
+    file = await wrap_async_op(StorageFile.GetFileFromPathAsync(os.fspath(file_path)))
```

Comment:
>wrap_async_op [](start = 17, length = 13)

is it possible to project each *Async winrt method as an "async def" so it can be awaited directly?  be nice to remove this boilerplate wrapping

Reply:
Yes, that's the plan - part of my next task. More details in #37